root/release_group/ReleaseGroupIndex.js

```diff
@@ -35,78 +35,85 @@ import {returnToCurrentPage} from '../utility/returnUri.js';
 
 import ReleaseGroupLayout from './ReleaseGroupLayout.js';
 
-function buildReleaseStatusTable(
-  $c: CatalystContextT,
-  releaseStatusGroup: $ReadOnlyArray<ReleaseT>,
-  releaseGroupCreditId: number | void,
-) {
-  const status = releaseStatusGroup[0].status;
-  return (
-    <React.Fragment key={status ? status.name : 'no-status'}>
-      <tr className="subh">
-        {$c.user ? <th /> : null}
-        <th colSpan={$c.session?.tport == null ? 8 : 9}>
-          {status?.name
-            ? lp_attributes(status.name, 'release_status')
-            : lp('(unknown)', 'release status')}
-        </th>
-      </tr>
-      {releaseStatusGroup.map((release, index) => (
-        <tr className={loopParity(index)} key={release.id}>
-          {$c.user
-            ? (
+function getReleaseStatusTableBuilder(showArtworkPresence: boolean) {
+  function buildReleaseStatusTable(
+    $c: CatalystContextT,
+    releaseStatusGroup: $ReadOnlyArray<ReleaseT>,
+    releaseGroupCreditId: number | void,
```

Comment:
Have you considered just adding a `showArtworkPresence` presence argument to the existing `buildReleaseStatusTable` function? I'm not quite sure I understand if there's a reason it can't be passed like this below, rather than having to create a new function for each invocation:

```JS
                  {releases.map(releaseStatusGroup => buildReleaseStatusTable(
                    $c,
                    releaseStatusGroup,
                    releaseGroup.artistCredit.id,
                    showArtworkPresence,
                  ))}
```

Reply:
In this case it's not necessary I think, but in the other files `buildResults` is passed as a function to a component, so there's no opportunity to give it arguments. I guess I got carried away repeating the pattern, I'll clean it up.

Reply:
Ah, I see. For `buildResults` I'd probably do the same, and just inline the builder:

```JS
const showArtworkPresence = results.some((res) => res.entity.hasCoverArt);
// [...]
        buildResult={(result, index) => buildResult(
          result,
          index,
          showArtworkPresence,
        )}
```
What you have is also fine, but the above would keep the diff smaller and reduce the amount of indentation needed, at least. :)

Reply:
I see! That's smart, something for me to remember! 😄 It does result in a slightly deeper call stack, but it's more understandable and concise